presto-main/src/main/java/com/facebook/presto/server/QuerySessionSupplier.java

```diff
@@ -128,6 +128,8 @@ public Session createSession(QueryId queryId, SessionContext context, Optional<S
                 .setClientTags(context.getClientTags())
                 .setResourceEstimates(context.getResourceEstimates());
 
+        context.getTraceToken().ifPresent(token -> sessionBuilder.setTraceToken(Optional.of(token)));
```

Comment:
This can be simplified to
```java
.setTraceToken(context.getTraceToken())
```
And moved to the chained set of calls above. (It's fine to set an empty optional into the builder.)